webapp/javascript/components/TimelineChart/ContextMenu.plugin.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import * as ReactDOM from 'react-dom';
+import { randomId } from '@webapp/util/randomId';
+import { Provider } from 'react-redux';
+import store from '@webapp/redux/store';
+
+// Pre calculated once
+// TODO(eh-am): does this work with multiple contextMenus?
+const WRAPPER_ID = randomId('contextMenu');
+
+export interface ContextMenuProps {
+  click: {
+    /** The X position in the window where the click originated */
+    pageX: number;
+    /** The Y position in the window where the click originated */
+    pageY: number;
+  };
+  timestamp: number;
+}
+
+(function ($: JQueryStatic) {
+  let globalPlot: jquery.flot.plot;
+
+  function onClick(
+    event: unknown,
+    pos: { x: number; pageX: number; pageY: number }
+  ) {
+    const container = inject($);
+    const containerEl = container?.[0];
+
+    // unmount any previous menus
+    ReactDOM.unmountComponentAtNode(containerEl);
+
+    // TODO(eh-am): improve typing
+    const ContextMenu = (globalPlot.getOptions() as ShamefulAny).ContextMenu as
+      | React.FC<ContextMenuProps>
+      | undefined;
+
+    if (ContextMenu) {
+      // TODO(eh-am): why do we need this conversion?
+      const timestamp = Math.round(pos.x / 1000);
```

Comment:
I would get rid of this conversion at all

`const timestamp = pos.x`

Reply:
the point here is to expose to `ContextMenu` a timestamp in the same format we passed (ie what we got from the backend), `ContextMenu` doesn't have to know about its underlying format

https://github.com/pyroscope-io/pyroscope/blob/main/webapp/javascript/components/TimelineChart/TimelineChartWrapper.tsx#L503